app/wizards/placements/add_placement/steps/additional_subjects.rb

```diff
@@ -0,0 +1,39 @@
+class Placements::AddPlacement::Steps::AdditionalSubjects
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  attribute :school
+  attribute :parent_subject_id
+  attribute :additional_subject_ids, default: []
+
+  validates :school, presence: true
+  validates :parent_subject_id, presence: true
+  validates :additional_subject_ids, presence: true
+  validate :additional_subjects_valid
+
+  delegate :name, to: :parent_subject, prefix: true, allow_nil: true
+
+  def additional_subjects_for_selection
+    parent_subject.child_subjects
+  end
+
+  def additional_subject_ids=(value)
+    super Array(value)
+  end
+
+  def wizard_attributes
+    { additional_subject_ids: }
+  end
+
+  def parent_subject
+    @parent_subject ||= Subject.find(parent_subject_id)
+  end
+
+  private
+
+  def additional_subjects_valid
+    return if additional_subject_ids.all? { |id| Subject.exists?(id:) }
```

Comment:
### Check that it's a child of the parent subject

I think we could tighten this up a little. We should probably be checking that all `additional_subject_ids` are children of `parent_subject`, rather than just checking they exist.

### Performance tweak

Also, it _might be_ more performant? 🤔 I think it should cut out N queries to the database by avoiding the loop.

```suggestion
return if (additional_subject_ids - parent_subject.child_subjects.ids).blank?
```

If every additional subject is a child of the parent, then we should end up with an empty array. Any outliers will remain. I appreciate the array mathematics might hurt readability, though.

### The `inclusion` validator

In fact, now I think about it, I think the `inclusion` validator should be able to handle it. And then we can drop the custom validation method altogether.

It turns out `inclusion` [accepts a lambda](https://edgeapi.rubyonrails.org/classes/ActiveModel/Validations/HelperMethods.html#method-i-validates_inclusion_of) 🙌🏻 which means it should be possible to do something like this:

```ruby
validates :additional_subject_ids, presence: true,
          inclusion: { in: ->(step) { step.additional_subjects_valid.ids } }
```